python/tests/test_tables.py

```diff
@@ -3753,6 +3758,113 @@ def verify_union_equality(self, tables, other, node_mapping, add_populations=Tru
         tables.subset(orig_nodes)
         tsutil.assert_table_collections_equal(uni1, tables, ignore_provenance=True)
 
+    def verify_union_consistency(self, tables, other, node_mapping):
+        ts1 = tsutil.insert_unique_metadata(tables)
+        ts2 = tsutil.insert_unique_metadata(other, offset=1000000)
+        for tsk in [False, True]:
```

Comment:
This is checking both the tskit union and py_union for consistency; we don't need to do that now that they both pass, I guess - but I did this to find the error in py_union.

Reply:
These are great and close #1243

Reply:
oh sweet

Reply:
To really close that one we should tests mutations and sites too; I've done this and updated.